agentlib/core/datamodels.py

```diff
@@ -640,6 +640,8 @@ def convert_to_pd_series(value):
         )
     if isinstance(srs.index[0], str):
         srs.index = srs.index.astype(float)
+    if isinstance(srs.index, pd.DatetimeIndex):
+        srs.index = pd.to_numeric(srs.index) / 10**9
```

Comment:
I solved the same issue doing this in AgentLib-MPC:

```
    if isinstance(trajectory, str):
        trajectory = pd.read_json(StringIO(trajectory), typ="series", convert_axes=False)
        trajectory.index = trajectory.index.astype(float)
```
I think it should be directly after the json parsing.